test/functional/test_framework/test_framework.py

```diff
@@ -879,9 +880,16 @@ def skip_if_no_py_sqlite3(self):
     def skip_if_no_py_capnp(self):
         """Attempt to import the capnp package and skip the test if the import fails."""
         try:
-            import capnp  # type: ignore[import] # noqa: F401
-        except ImportError:
-            raise SkipTest("capnp module not available.")
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
```

Comment:
In commit "test: Ignore error message give from python because of GIL" (136dfde1c2284e30ecb1dfb35520f93878ec7335)

Could this change to test_framework.py be reverted? 

I would think since the capnp module will already be imported by interface_ipc.py before this runs, it's impossible for the warning to ever trigger here.

If changing this method is actually necessary, I think it'd be good to move common code importing capnp and suppressing the warning to a function in test_framework/util.py that can do this and return a reference to the capnp module. This way the catch/filter/import code doesn't need to be repeated.